### ISSUE TITLE
1.21.1 Many failing tests on py310 and py311 (AttributeError: 'SwigPyObject' object has no attribute 'thisown')

### ISSUE THREAD

**Issue by dvzrv** [pymupdf/PyMuPDF#2328]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
I'm currently doing rebuilds for Arch Linux as we are updating to Python 3.11.
There are dozens of failing tests rebuilding this project against Python 3.10 and 3.11.
All relate to `AttributeError: 'SwigPyObject' object has no attribute 'thisown'`.

The tests used to work when I built the package last (2022-12-15). I assume that e.g. changes to swig or other components lead to these issues.

## To Reproduce (mandatory)

```shell
# building against mupdf 1.21.1
export PYMUPDF_SETUP_MUPDF_BUILD=""
python -m build --wheel --no-isolation
python -m installer --destdir=test_dir dist/*.whl
export PYTHONPATH="test_dir/$_site_packages:$PYTHONPATH"
# ignore other failing tests (see #2040)
pytest -vv -c /dev/null tests/ -k 'not test_textbox3 and not test_color_count'
```

Python 3.10:
[python-pymupdf-1.21.1-2-x86_64-build.log](https://github.com/pymupdf/PyMuPDF/files/11185296/python-pymupdf-1.21.1-2-x86_64-build.log)
[python-pymupdf-1.21.1-2-x86_64-check.log](https://github.com/pymupdf/PyMuPDF/files/11185297/python-pymupdf-1.21.1-2-x86_64-check.log)

Python 3.11:
[python-pymupdf-1.21.1-2-x86_64-build.log](https://github.com/pymupdf/PyMuPDF/files/11185300/python-pymupdf-1.21.1-2-x86_64-build.log)
[python-pymupdf-1.21.1-2-x86_64-check.log](https://github.com/pymupdf/PyMuPDF/files/11185301/python-pymupdf-1.21.1-2-x86_64-check.log)

## Expected behavior (optional)
Tests do not fail.

## Screenshots (optional)
n/a

## Your configuration (mandatory)
 - Arch Linux x86_64
 - Python 3.10.10 and 3.11.3
 - PyMuPDF 1.21.1 (source tarball)

**Comment by julian-smith-artifex-com** [comment]
Interesting, i'm not sure what's going on.

The only thing i can think of is that somehow the runtime version of MuPDF is different from the one that PyMuPDF is built with.

It's late here, but tomorrow i'll try replicating your setup - building from PyMuPDF-1.21.1 source tarball with MuPDF-1.21.1 source.

**Comment by julian-smith-artifex-com** [comment]
I've reproduced the problem on my machine.

Unfortunately it looks like new builds of PyMuPDF-1.21.1/MuPDF-1.21.1 (released 2022-12-13) are not working on Linux. As you suggest, it might be something to do with newer versions of swig. For me the PyMuPDF-1.21.1 Linux wheel on pypi.org passes its own tests ok, but fails some tests with SEGVs when used with the current PyMuPDF test suite, which suggests that there are bugs in PyMuPDF-1.21.1 that are exposed by more recent swig code.

Using the current PyMuPDF (1.21 branch) with MuPDF-1.21.1 is tested daily on Github and works fine. There have been fixes for SEGVs,  and in particular a bisect shows that commit https://github.com/pymupdf/PyMuPDF/commit/e1b6e5cc3724 #e1b6e5cc3724 "fix possible segvs in Page.reload_page() and Annot.delete_annot()." is the critical one for making all tests pass on Linux.

The current PyMuPDF (1.21 branch) is essentially ready to go as a new release PyMuPDF-1.22.0; we're currently waiting for the corresponding MuPDF-1.22.0 to be released, which should be sometime this week.

PyMuPDF (1.21 branch) will work with both MuPDF-1.21.1 and the upcoming MuPDF-1.22.0, so could you try building with the latest PyMuPDF (1.21 branch) and your system-installed MuPDF?

**Comment by dvzrv** [comment]
Only applying e1b6e5cc37247e204ae54251fa68104384504098 is not sufficient it seems (same issues).

Building from https://github.com/pymupdf/PyMuPDF/commit/8d05ff14bc9fc4223116c85efb843d896203b329 the build is broken and swig is allegedly not found (but it is installed):

```
* Getting build dependencies for wheel...
python-pymupdf/setup.py: sys.argv: ['/usr/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py', 'egg_info']
python-pymupdf/setup.py: os.getcwd(): /build/python-pymupdf/src/python-pymupdf
python-pymupdf/setup.py: __file__: /build/python-pymupdf/src/python-pymupdf/setup.py
python-pymupdf/setup.py: $PYTHON_ARCH: None
python-pymupdf/setup.py: os.environ (32):
python-pymupdf/setup.py:     SHELL: /bin/bash
python-pymupdf/setup.py:     BUILDTOOL: devtools
python-pymupdf/setup.py:     SUDO_GID: 0
python-pymupdf/setup.py:     PYTHONHASHSEED: 0
python-pymupdf/setup.py:     TEXTDOMAINDIR: /usr/share/locale
python-pymupdf/setup.py:     SUDO_COMMAND: /bin/bash -c bash -c cd\ /startdir;\ makepkg\ "$@" -bash --syncdeps --noconfirm --log --holdver --skipinteg --install
python-pymupdf/setup.py:     SUDO_USER: root
python-pymupdf/setup.py:     PWD: /build/python-pymupdf/src/python-pymupdf
python-pymupdf/setup.py:     SOURCE_DATE_EPOCH: 1681239884
python-pymupdf/setup.py:     LOGNAME: builduser
python-pymupdf/setup.py:     CXXFLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/python-pymupdf/src=/usr/src/debug/python-pymupdf -flto=auto
python-pymupdf/setup.py:     COMMAND_MODE: legacy
python-pymupdf/setup.py:     LDFLAGS: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -flto=auto
python-pymupdf/setup.py:     HOME: /build
python-pymupdf/setup.py:     LANG: C.UTF-8
python-pymupdf/setup.py:     RUSTFLAGS:  -C debuginfo=2 --remap-path-prefix=/build/python-pymupdf/src=/usr/src/debug/python-pymupdf
python-pymupdf/setup.py:     PYMUPDF_SETUP_MUPDF_BUILD:
python-pymupdf/setup.py:     MAKEFLAGS: -j24
python-pymupdf/setup.py:     TERM: tmux-256color
python-pymupdf/setup.py:     USER: builduser
python-pymupdf/setup.py:     SHLVL: 1
python-pymupdf/setup.py:     DEBUGINFOD_URLS: https://debuginfod.archlinux.org
python-pymupdf/setup.py:     CHOST: x86_64-pc-linux-gnu
python-pymupdf/setup.py:     PATH: /usr/local/sbin:/usr/local/bin:/usr/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl
python-pymupdf/setup.py:     CFLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/python-pymupdf/src=/usr/src/debug/python-pymupdf -flto=auto
python-pymupdf/setup.py:     SUDO_UID: 0
python-pymupdf/setup.py:     MAIL: /var/mail/builduser
python-pymupdf/setup.py:     BUILDTOOLVER: 20230307-1-any
python-pymupdf/setup.py:     OLDPWD: /build/python-pymupdf/src
python-pymupdf/setup.py:     TEXTDOMAIN: pacman-scripts
python-pymupdf/setup.py:     _: /usr/bin/python
python-pymupdf/setup.py:     PEP517_BUILD_BACKEND: setuptools.build_meta
python-pymupdf/setup.py: platform.system()='Linux'
python-pymupdf/setup.py: sys.platform='linux'
python-pymupdf/setup.py: include_dirs=[]
python-pymupdf/setup.py: library_dirs=[]
python-pymupdf/setup.py: libraries=[]
python-pymupdf/setup.py: extra_swig_args=['-w509']
python-pymupdf/setup.py: extra_compile_args=[]
python-pymupdf/setup.py: extra_link_args=[]
running egg_info
creating PyMuPDF.egg-info
writing PyMuPDF.egg-info/PKG-INFO
writing dependency_links to PyMuPDF.egg-info/dependency_links.txt
writing top-level names to PyMuPDF.egg-info/top_level.txt
writing manifest file 'PyMuPDF.egg-info/SOURCES.txt'
file fitz/fitz.py (for module fitz.fitz) not found
reading manifest file 'PyMuPDF.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'mupdf.tgz'
warning: no previously-included files matching '__pycache__/*' found anywhere in distribution
adding license file 'COPYING'
writing manifest file 'PyMuPDF.egg-info/SOURCES.txt'

ERROR Missing dependencies:
        swig
```

This is with swig 4.1.1 and I have all the other build requirements in (just like before):

```
export PYMUPDF_SETUP_MUPDF_BUILD=""
python -m build --wheel --no-isolation
```

**Comment by julian-smith-artifex-com** [comment]
BTW we've just released PyMuPDF-1.22.1. This has a fix for #2355 (https://github.com/pymupdf/PyMuPDF/commit/f82c9be5eacab) that might be relevant here.

#2355 involves error messages like `AttributeError: 'SwigPyObject' object has no attribute`; it looks like it only happens with swig-4.1.1.

One other thing: code for 1.22.1 is on new branch `main`.
